README.md

```diff
@@ -239,24 +239,463 @@ One can also do it manually:
+
+#### ❌ DONT move the `module State`, or `evolve` logic out
+
+The State type and the associated `evolve` and `fold` functions are intimately tied to the Event Contracts. Over time, ugliness and upconversion can lead to noise, and temptation to move it out. Don't do it; being able to understand the full coupling is critical to understanding how things work, and equally critical to being able to change or add functions.
+
+#### ❌ DONT move the decision logic out
+
+Decision logic bridges between the two worlds of State and Events. The State being held exists only to serve the Decision logic. The only reason for Event Contracts is to record Decisions. Trying to pretend that some of the Decisions are less important and hence should live elsewhere is rarely a good idea. How decisions are made, and how those decisions are encoded as Events should be encapsulated within the Aggregate.
+
+### 2. `module Events`
+
+Having the Event Contracts be their own `module` is a critical forcing function for good aggregate design. Having all types and all cases live in one place and being able to quickly determine where each Event is produced is key to being able to understand the moving parts of a system.
+
+<a name="events-no-ids"></a>
+#### ❌ AVOID including egregious identity information
+
+When modelling, it's common to include primary identifiers (e.g. a user id), or contextual identifiers (e.g. a tenant id) in an Event in order to convey the relationships between events in the systems as a whole; you want the correlations to stand out. In the implementation however, repeating the identity information in every event is a major liability:
+1. the State needs to contain the values - that's more noise
+2. Event versioning gets messier - imagine extending a system to make it multi-tenant, you'd need to be able to handle all the historic events that predated the concept
+
+The alternative is for a workflow to react to the events in the context of a stream - if some logic needs to know the userid let the User reactor handing the User event on a User Stream pass thgat context forward if relevant in that context.
+
+#### ❌ DONT `open Events` in an aggregate module
+
+Having to prefix types and/or Event Type names with `Events.` is a feature, not a bug. 
+
+### 3. `module Fold`
+
+<a name="fold-no-log"></a>
+#### ❌ DONT log
+
+If Fold logic is not incredibly boring, that's a design smell. In general, you want to [make illegal States unrepresentable](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/). If you must, unit test it to satisfy yourself things can't go wrong.
```

Comment:
> If Fold logic is not incredibly boring, that's a design smell

Reword, "If Fold logic is complicated" or "If Fold logic is anything but incredibly boring"

Took a while to parse due to the double negative

Reply:
https://github.com/jet/dotnet-templates/pull/133/commits/87a764c2e18a7a62c693a020f806d1c9b0af0f3a